pkgs/by-name/up/upscaler/package.nix

```
@@ -0,0 +1,72 @@
+{
+  lib,
+  fetchFromGitLab,
+  meson,
+  ninja,
+  gitUpdater,
+  desktop-file-utils,
+  appstream,
+  blueprint-compiler,
+  python3Packages,
+  pkg-config,
+  libadwaita,
+  wrapGAppsHook4,
+  upscayl-ncnn,
+}:
+python3Packages.buildPythonApplication rec {
+  pname = "upscaler";
+  version = "1.4.0";
+
+  pyproject = false; # meson instead of pyproject
+
+  src = fetchFromGitLab {
+    domain = "gitlab.gnome.org";
+    owner = "World";
+    repo = "Upscaler";
+    rev = version;
+    hash = "sha256-Dy8tykIbK5o0XulurG+TxORZZSxfRe5Kjh6aGpsh+0Y=";
+  };
+
+  passthru.updateScript = gitUpdater { };
+
+  postPatch = ''
+    substituteInPlace upscaler/window.py \
+      --replace-fail '"upscayl-bin",' '"${lib.getExe upscayl-ncnn}",'
+  '';
+
```

```suggestion
  strictDeps = true;
```
Not required, but nice for this case as we end up dealing with both buildPlatform and hostPlatform versions of the same packages (like gtk)